op-acceptance-tests/cmd/flake-shake-promoter/main.go

```diff
@@ -828,6 +864,71 @@ func listJobs(ctx *apiCtx, workflowID string) (jobList, error) {
 	return jl, nil
 }
 
+// resolveReportArtifactsURL attempts to find the web URL to the report job's artifacts page
+// by scanning recent pipelines/workflows for the configured workflow/report job names.
+// Returns an empty string if not found.
+func resolveReportArtifactsURL(opts promoterOpts, ctx *apiCtx) string {
+	// Scan the latest pipelines on the given branch; reuse collectReports traversal but short-circuit on first match
+	basePipelines := fmt.Sprintf("https://circleci.com/api/v2/project/gh/%s/%s/pipeline?branch=%s", url.PathEscape(opts.org), url.PathEscape(opts.repo), url.QueryEscape(opts.branch))
+	pageURL := basePipelines
+	now := time.Now().UTC()
+	since := now.AddDate(0, 0, -opts.daysBack)
+	for {
+		pl, nextToken, err := getPipelinesPage(ctx, pageURL)
+		if err != nil {
+			return ""
+		}
+		for _, p := range pl.Items {
+			if p.CreatedAt.Before(since) {
+				return ""
+			}
+			wfl, err := listWorkflows(ctx, p.ID)
+			if err != nil {
+				return ""
+			}
+			for _, w := range wfl.Items {
+				if w.Name != opts.workflowName {
+					continue
+				}
+				jl, err := listJobs(ctx, w.ID)
+				if err != nil {
+					return ""
+				}
+				for _, j := range jl.Items {
+					if j.Name != opts.reportJobName {
+						continue
+					}
+					if j.WebURL != "" {
+						url := j.WebURL
+						if !strings.Contains(url, "/artifacts") {
+							if strings.HasSuffix(url, "/") {
+								url = url + "artifacts"
+							} else {
+								url = url + "/artifacts"
+							}
+						}
+						return url
```

Comment:
I guess the usage of this is limited, looks like a very fragile URL manipulation (e.g. query string or hash will break this)

Reply:
True. I'll try circle back to this and do proper URL-aware manipulation. But I doubt it will break in practise.